modules/imgcodecs/src/ios_conversions.mm

```diff
@@ -116,6 +118,8 @@ void UIImageToMat(const UIImage* image,
         if (!alphaExist)
             bitmapInfo = kCGImageAlphaNoneSkipLast |
                                 kCGBitmapByteOrderDefault;
+        else
+            m = cv::Scalar(0);
```

Comment:
I considered that.  I chose not to so for efficiency reasons since setting all those bytes takes time and this function is called to process frames in video streams in some use cases (like my current one).  Since the alpha channel is ignored it seemed like a reasonable efficiency tradeoff.